docs/core/profiles.md

```diff
@@ -139,108 +141,152 @@ And there are others that take on a numeric value, like:
 - max_weight
 - max_width
 
-*Important note: Whenever you want to use any of these categories for a custom profile you need to add them to 
+*Important note: Whenever you want to use any of these categories for a custom profile you need to add them to
 `graph.encoded_values` in `config.yml`.*
 
 ### Setting up a Custom Model
 
 As mentioned above, the custom weighting function has three parameters that you can adjust: speed, priority and
 distance_influence. You can set up rules that determine these parameters from the edge's properties. A set of such rules
-is called a 'custom model' and it is written in a dedicated YAML format. We will now see how the cost function parameters
-can be influenced by the different fields of such a custom model.
+is called a 'custom model' and it is written in a dedicated YAML or JSON format. We will now see how the cost function 
+parameters can be influenced by the different fields of such a custom model.
   
 #### Customizing `speed`
 
-For every edge a default speed is inherited from the base vehicle, but you have multiple options to adjust it.
-The first thing you can do is rescaling the default speeds using the `speed_factor` section. For example this is how you
-can reduce the the speed of every edge that has the value 'motorway' for the category 'road_class' to fifty percent of 
+For every road segment a default speed is inherited from the base vehicle and you have multiple options to adjust it.
+The first thing you can do is rescaling the default speeds using `multiply by`. For example this is how you
+can reduce the speed of every road segment that has the value `MOTORWAY` for the category 'road_class' to fifty percent of 
 the default speed that is normally used by the base vehicle for this road class:
 ```yaml
-speed_factor:
-  road_class: {motorway: 0.5}
-```  
-Note that `road_class: {motorway: 0.5}` is an alternative YAML notation that is equivalent to:
-```yaml
-speed_factor:
-  road_class:
-    motorway: 0.5
+speed:
+  - if: road_class == MOTORWAY
+    multiply by: 0.5
 ```
 
-You can also setup speed factors for multiple road classes like this
+You can also change the speed in case of different road classes like this
 ```yaml
-speed_factor:
-  road_class: {motorway: 0.5, primary: 0.7, tertiary: 0.9}
+speed:
+  - if: road_class == MOTORWAY
+    multiply by: 0.5
+  - if: road_class == PRIMARY || road_class == TERTIARY
+    multiply by: 0.7
 ```
 
-and use multiple categories to influence the speed factor
+The OR-operator `||` means if the left **or** right expressions is true then the value `0.7` is used.
+You can also use multiple categories in the expressions:
+
 ```yaml
-speed_factor:
-  road_class: {motorway: 0.5}
-  road_environment: {tunnel: 0.8}
+speed:
+  - if: "road_class == MOTORWAY",     multiply by: 0.5
+  - if: "road_environment == TUNNEL", multiply by: 0.8
```

Comment:
Maybe we should only use the same notation everywhere here and instead add a separate section that explains how one can also use this more JSON style notation (after the reader understood the yaml format). I found it distracting to see two different notations here.

Reply:
One thing that bothers me a bit is that more (?) users will do client side custom models than server side ones and the current profiles.md (after my edits) focuses on the `yaml` format even though for cusom-route requests you need JSON. But really that should not be too hard to translate?

Reply:
Hmmh, yes. I created a follow up issue where we clarify this and some more points: https://github.com/graphhopper/graphhopper/issues/2232